Fix __repr__ crash in Python 3

lwneal opened the pull request:
In Python >3.0, the following results in a TypeError:

````
    >>> import libmr
    >>> mr = libmr.MR()
    >>> print(mr)
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      TypeError: __str__ returned non-string (type bytes)
````

This patch restores the expected behavior of converting to str.

````
>>> import libmr
>>> libmr.MR()
<MR object: ''>
````
Tested on Python 2.7.12 and 3.5.2. Behavior in Python 2 remains the same.

siebenkopf commented:
Lawrence, thanks for providing a patch. 

Note that the `to_string` function is supposed to be used for reading and writing `MR` objects to (text) files. Have you checked that the `load_from_string` function will still work with your modification?

```
>>> import libmr
>>> mr = libmr.MR()
>>> mr2 = libmr.load_from_string(str(mr))
```

lwneal commented:
Thanks for the review. Although this patch fixed serializing with `to_string`, it looks like deserializing with `load_from_string` was still broken.

I've updated the patch now with an additional change to fix `load_from_string`. To ensure that it works, I also added a simple `test.py` in examples.

Output of the test in both Python 3.5 and 2.7 on my system is:
````
Ran 2 tests in 0.000s

OK
````
